dask/blockwise.py

```diff
@@ -256,6 +263,17 @@ def __init__(
             self.concatenate = concatenate
         self.new_axes = new_axes or {}
 
+    @property
+    def dims(self):
+        """Returns a dictionary mapping between each index specified in
+        `self.indices` and the number of output blocks for that indice.
+        """
+        if not hasattr(self, "_dims"):
+            self._dims = broadcast_dimensions(self.indices, self.numblocks)
+            for k, v in self.new_axes.items():
+                self._dims[k] = len(v) if isinstance(v, tuple) else 1
```

Comment:
This logic now exists in `make_blockwise_graph` too, do you think it's worth moving to a separate function?